cpp/bench/matrix/select_k.cu

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../../test/matrix/select_k.cuh"
```

Comment:
Can you please reference issue https://github.com/rapidsai/raft/issues/1153 here? I'm okay having code shared between bench and test but I'd like to avoid having to link against gtest in benchmarks.

Reply:
Ok, though this file does not depend on gtest.

Reply:
We do have a file which caused the error in #1153 and my main gripe with using this pattern where we're sharing these headers between `bench/` and `test/` is that it's too easy to do this by mistake and cause build errors which won't be caught in CI. Sure, we could just link against gtest in bench (which is what we've done for the near term) but I think it's a sign of a larger bad smell that we could alleviate by putting these headers somewhere else. I'm not sure where that location would be, though. Maybe in `include/raft/test` or maybe follow the general pattern of the `specializations/` directories and keep the headers in a known place within each namespace (e.g. `include/raft/matrix/test`) ? I suppose another option could be to add an additional `include` directory like `include/raft_test`. I'm sure there's other options too but it would be nice to see these types of shared includes moved out of `test/` before we create too many more of them.

Reply:
Sure, I get it! What about something like "include_internal", "include_test", or "internal" at the root of "cpp" folder? To place all headers that can be shared between test and bench.